src/main/java/org/jenkinsci/plugins/ssegateway/Endpoint.java

```diff
@@ -124,15 +135,19 @@ public HttpResponse doConnect(StaplerRequest request, StaplerResponse response)
         // If there was already a dispatcher with this ID, then remove
         // all subscriptions from it and reuse the instance.
         if (dispatcher != null) {
-            LOGGER.log(Level.INFO, "We already have a Dispatcher for clientId {0}. Removing all subscriptions on the existing Dispatcher instance and reusing it.", dispatcher.toString());
+            LOGGER.log(Level.FINE, "We already have a Dispatcher for clientId {0}. Removing all subscriptions on the existing Dispatcher instance and reusing it.", dispatcher.toString());
             dispatcher.unsubscribeAll();
         } else {
             // Else create a new instance with this id.
             EventDispatcherFactory.newDispatcher(clientId, session);
         }
         
         response.setStatus(HttpServletResponse.SC_OK);
-        return HttpResponses.okJSON(); 
+        
+        JSONObject responseData = new JSONObject();
+        responseData.put("jsessionid", session.getId());
```

Comment:
i.e. that thing makes testing etc (via the headless client) a lot easier so I want to keep it.